src/Common/Menus/Traits/With_Admin_Bar.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Adds menu item to the WP admin bar.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Menus
+ */
+
+namespace TEC\Common\Menus\Traits;
+
+trait With_Admin_Bar {
+
+	/**
+	 * slug of the adminbar parent menu.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $adminbar_parent = '';
+
+	public function hooks() {
```

Comment:
Too generic a method name for a trait, conflicts will arise.

Reply:
I agree (and have had said collisions 😜) - I am working on that